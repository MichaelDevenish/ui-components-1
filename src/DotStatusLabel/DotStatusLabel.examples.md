## Basic label

```jsx
<DotStatusLabel
  text='Hello world!'
/>
```

## Basic label with subtitle text

```jsx
<DotStatusLabel
  text='Hello world!'
  subtitleText='FooBar'
/>
```

## Hollow label

```jsx
<DotStatusLabel
  text='Hello world!'
  isHollow
/>
```

## Coloured text

```jsx
<DotStatusLabel
  text='Hello world!'
  subtitleText='FooBar'
  textColor='success'
/>
```

## Coloured icon

```jsx
<div>
  <DotStatusLabel
    text='Hello world!'
    subtitleText='FooBar'
    color='rgb(128, 0, 96)'
  />

  <DotStatusLabel
    text='Hello world!'
    subtitleText='FooBar'
    color='rgb(0, 96, 128)'
    isHollow
  />
</div>
```
