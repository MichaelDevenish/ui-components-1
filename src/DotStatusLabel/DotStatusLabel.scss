:local(.DotStatusLabel) {
  font-size: .9375rem;
  font-weight: normal;
  vertical-align: bottom;

  .fa {
    bottom: 2px;
    font-size: xx-small;
    position: relative;
  }

  .subtitle {
    font-size: .8125rem;
    padding-left: 15px;
  }

  @each $color, $value in $functional-colours {
    &.#{$color} {
      color: map-get($value, base)
    }
  }
}
