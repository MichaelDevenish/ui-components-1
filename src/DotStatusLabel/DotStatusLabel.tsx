import classNames from 'classnames'
import React from 'react'
import { Icon } from '../Icon'

const style = require('./DotStatusLabel.scss')

export interface DotStatusLabelProps {
    /** Text to show inside the label  */
    text: string

    /** Smaller text shown under the label */
    subtitleText?: string

    /** Colour of the icon  */
    color?: string

    /** Main colour of the text (and icon if no `color` provided)  */
    textColor?: 'alert' | 'success' | 'warning' | 'primary' | 'neutral' | 'secondary'

    /** Should the label use a hollow icon  */
    isHollow?: boolean
  }

export class DotStatusLabel extends React.Component<DotStatusLabelProps> {
  public static defaultProps: Partial<DotStatusLabelProps> = {
    isHollow: false
  }

  get subtitleText (): JSX.Element | null {
    const {
      subtitleText,
      textColor
    } = this.props

    if (subtitleText) {
      return (
        <div className={classNames('subtitle', textColor)}>
          {subtitleText}
        </div>
      )
    }

    return null
  }

  public render (): JSX.Element {
    const {
      text,
      color,
      textColor,
      isHollow
    } = this.props

    return (
      <span className={classNames(style.DotStatusLabel, 'dot-status-label', textColor)}>
        <Icon
          type={isHollow ? 'circle-o' : 'circle'}
          size={1}
          color={color}
        />
        {text}
        {this.subtitleText}
      </span>
    )
  }
}
