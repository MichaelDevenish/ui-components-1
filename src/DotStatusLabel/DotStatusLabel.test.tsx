import { shallow } from 'enzyme'
import React from 'react'

import { DotStatusLabel } from './DotStatusLabel'

describe('<DotStatusLabel />', () => {
  it('should render a dot status label with default values', () => {
    const wrapper = shallow(
      <DotStatusLabel
        statusText='Test'
      />
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('should render a dot status label', () => {
    const wrapper = shallow(
      <DotStatusLabel
        statusText='Test'
        color='rgb(192, 48, 0)'
      />
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('should render a hollow dot status label', () => {
    const wrapper = shallow(
      <DotStatusLabel
        statusText='Test'
        isHollow
      />
    )

    expect(wrapper).toMatchSnapshot()
  })
})
