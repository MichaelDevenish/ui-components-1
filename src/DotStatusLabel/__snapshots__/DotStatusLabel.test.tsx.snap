// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DotStatusLabel /> should render a dot status label 1`] = `
<span
  className="DotStatusLabel dot-status-label"
>
  <Icon
    color="rgb(192, 48, 0)"
    isFontAwesome={true}
    isLarge={false}
    isSpinning={false}
    isStacked={false}
    size={1}
    type="circle"
  />
</span>
`;

exports[`<DotStatusLabel /> should render a dot status label with default values 1`] = `
<span
  className="DotStatusLabel dot-status-label"
>
  <Icon
    isFontAwesome={true}
    isLarge={false}
    isSpinning={false}
    isStacked={false}
    size={1}
    type="circle"
  />
</span>
`;

exports[`<DotStatusLabel /> should render a hollow dot status label 1`] = `
<span
  className="DotStatusLabel dot-status-label"
>
  <Icon
    isFontAwesome={true}
    isLarge={false}
    isSpinning={false}
    isStacked={false}
    size={1}
    type="circle-o"
  />
</span>
`;
